src/txHashBuilder.c

```diff
@@ -127,7 +203,16 @@ void txHashBuilder_init(
 		builder->includeNetworkId = includeNetworkId;
 		if (includeNetworkId) numItems++;
 
-		ASSERT((3 <= numItems) && (numItems <= 13));
+		builder->includeCollateralReturn = includeCollateralReturn;
+		if (includeCollateralReturn) numItems++;
+
+		builder->includeTotalCollateral = includeTotalCollateral;
+		if (includeTotalCollateral) numItems++;
+
+		builder->remainingReferenceInputs = numReferenceInputs;
+		numItems++; // an array that is always included (even if empty)
```

Comment:
No, take a look at https://cips.cardano.org/cips/cip21/:
"Unless mentioned otherwise in this CIP, optional empty lists and maps must not be included as part of the transaction body or its elements."

We don't allow empty arrays for optional elements (cf. `numCollaterals`).